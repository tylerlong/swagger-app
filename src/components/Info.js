import React from 'react'
import R from 'ramda'
import { Form } from 'antd'
import { connect } from 'react-redux'

import InfoFormItem from './InfoFormItem'

class Info extends React.Component {
  render () {
    return (
      <div>
        <h2>Info</h2>
        <Form>
          {['title', 'version', 'description', 'termsOfService', 'host', 'basePath', 'schemes', 'produces', 'consumes'].map(path => <InfoFormItem path={path} key={path} />)}
        </Form>
      </div>
    )
  }
}

export default connect(R.identity)(Info)
