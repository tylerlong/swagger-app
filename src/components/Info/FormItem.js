import React from 'react'
import R from 'ramda'
import { connect } from 'react-redux'
import { Input, Form } from 'antd'

import { setProp } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component {
  render () {
    console.log(`render Info.FormItem`)
    const { name, info, setProp } = this.props
    const value = info[name]
    return (
      <Form.Item {...formItemLayout} label={name}>
        <Input placeholder={name} size='large' value={value} onChange={(event) => { setProp(['info', name], event.target.value) }} />
      </Form.Item>
    )
  }
}

export default connect(R.pick(['info']), { setProp })(FormItem)
