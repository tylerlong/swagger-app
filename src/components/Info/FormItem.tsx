import React from 'react'
import R from 'ramda'
import { connect } from 'react-redux'
import { Input, Form } from 'antd'

import { setProp } from '../../actions'
import { formItemLayout } from '../../utils'

class FormItem extends React.Component<any> {
  render() {
    console.log(`render Info.FormItem`)
    const { name, value, setProp } = this.props
    let input = <Input placeholder={name} size='large' value={value} onChange={(event: any) => { setProp(name, event.target.value) }} />
    if (R.contains(name, ['schemes', 'produces', 'consumes'])) {
      input = <Input placeholder='Values separated by commas' size='large' defaultValue={R.join(', ', value)} onChange={(event: any) => { setProp(name, R.pipe(R.split(','), R.map(R.trim), R.reject(R.equals('')))(event.target.value)) }} />
    }
    return (
      <Form.Item {...formItemLayout} label={name}>
        {input}
      </Form.Item>
    )
  }
}

const mapStateToProps = ({ info }, { name }) => ({ value: info[name] })
const mapDispatchToProps = (dispatch) => ({
  setProp: (key, value) => dispatch(setProp(['info', key], value))
})
export default connect(mapStateToProps, mapDispatchToProps)(FormItem)
