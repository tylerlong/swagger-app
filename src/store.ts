import { createStore, applyMiddleware } from 'redux'
import { createLogicMiddleware } from 'redux-logic'

import reducer from './reducers'
import logics from './logics'

const logicMiddleware = createLogicMiddleware(logics)
const store = createStore(
  reducer,
  applyMiddleware(logicMiddleware)
)

export default store
