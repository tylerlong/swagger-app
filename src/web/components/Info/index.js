import React from 'react'

import InfoTextField from '../../containers/Info/InfoTextField'

class Info extends React.Component {
  render () {
    console.log(`render Info`)
    return (
      <div>
        <h2>Info</h2>
        <InfoTextField name='title' />
        <InfoTextField name='version' />
        <InfoTextField name='description' />
        <InfoTextField name='termsOfService' />
        <InfoTextField name='host' />
        <InfoTextField name='basePath' />
        <InfoTextField name='schemes' isArray />
        <InfoTextField name='produces' isArray />
        <InfoTextField name='consumes' isArray />
      </div>
    )
  }
}

export default Info
