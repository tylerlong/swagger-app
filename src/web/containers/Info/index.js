import { connect } from 'react-redux'

import { setProp } from '../../actions'
import TextField from '../../components/Common/TextField'

const mapStateToProps = ({ info }, { name }) => ({ value: info[name] })
const mapDispatchToProps = (dispatch, { name }) => ({
  update: value => dispatch(setProp(['info', name], value))
})
export const InfoTextField = connect(mapStateToProps, mapDispatchToProps)(TextField)
