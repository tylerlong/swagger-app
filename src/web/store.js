import * as R from 'ramda'
import { createStore, applyMiddleware } from 'redux'
import { createLogicMiddleware } from 'redux-logic'
import Rx from 'rxjs/Rx'
import { Base64 } from 'js-base64'

import reducer from './reducers'
import logics from './logics'
import { toSwagger } from './utils'

const logicMiddleware = createLogicMiddleware(logics)
const store = createStore(
  reducer,
  applyMiddleware(logicMiddleware)
)

// auto persist state
if (global.electron) {
  let persistedData
  const subject = new Rx.Subject().debounceTime(1000)
  subject.subscribe(() => {
    const state = R.omit(['alerts'], store.getState())
    const data = JSON.stringify(toSwagger(state), null, 2)
    if (data !== persistedData) {
      const tokens = window.location.href.split('#/edit/')
      if (tokens.length > 1) {
        const filePath = Base64.decode(tokens[1])
        global.fs.writeFileSync(filePath, data)
        persistedData = data
      }
    }
  })
  store.subscribe(() => {
    subject.next()
  })
}

export default store
