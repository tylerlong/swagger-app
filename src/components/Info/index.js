import React from 'react'
import R from 'ramda'
import { Form } from 'antd'
import { connect } from 'react-redux'

import FormItem from './FormItem'

class Info extends React.Component {
  render () {
    return (
      <div>
        <h2>Info</h2>
        <Form>
          {['title', 'version', 'description', 'termsOfService', 'host', 'basePath', 'schemes', 'produces', 'consumes'].map(path => <FormItem path={path} key={path} />)}
        </Form>
      </div>
    )
  }
}

export default connect(R.identity)(Info)
