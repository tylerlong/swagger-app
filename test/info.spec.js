/* eslint-env jest */
import R from 'ramda'

import Info from '../src/web/components/Info'
import store from './store'
import { getWrapper } from './shared'
import state from '../dist/state.json'

let wrapper = null
beforeEach(() => {
  wrapper = getWrapper(Info, R.pick(['info'], state))
})

describe('test info', () => {
  test('view info', () => {
    expect(wrapper.find('input').first().props().value).toEqual(store.getState().info.title)
  })
  test('update title', () => {
    wrapper.find('input').first().simulate('change', { target: { value: 'Hello' } })
    expect(store.getState().info.title).toEqual('Hello')
    wrapper.find('input').first().simulate('change', { target: { value: 'World' } })
    expect(store.getState().info.title).toEqual('World')
  })
  test('update arrays', () => {
    // schemes
    let input = wrapper.find('input[placeholder="Values separated by commas"]').first()
    input.simulate('change', { target: { value: 'https,http' } })
    expect(store.getState().info.schemes).toEqual(['https', 'http'])

    // produces
    input = wrapper.find('input[placeholder="Values separated by commas"]').at(1)
    input.simulate('change', { target: { value: 'application/json,text/plain; charset=utf-8' } })
    expect(store.getState().info.produces).toEqual(['application/json', 'text/plain; charset=utf-8'])

    // consumes
    input = wrapper.find('input[placeholder="Values separated by commas"]').at(2)
    input.simulate('change', { target: { value: 'application/json,text/plain; charset=utf-8' } })
    expect(store.getState().info.consumes).toEqual(['application/json', 'text/plain; charset=utf-8'])
  })
})
