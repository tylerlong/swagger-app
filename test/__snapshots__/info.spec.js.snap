// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test info view info 1`] = `
<Provider
  store={
    Object {
      "actions": Array [],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Info>
    <div>
      <h2>
        Info
      </h2>
      <Connect(FormItem)
        name="title"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="title"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="title"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="title"
                    >
                      title
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="title"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="Example API"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="title"
                          size="large"
                          type="text"
                          value="Example API"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="version"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="version"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="version"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="version"
                    >
                      version
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="version"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="1.0"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="version"
                          size="large"
                          type="text"
                          value="1.0"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="description"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="description"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="description"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="description"
                    >
                      description
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="description"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="Restful API for Example"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="description"
                          size="large"
                          type="text"
                          value="Restful API for Example"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="termsOfService"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="termsOfService"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="termsOfService"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="termsOfService"
                    >
                      termsOfService
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="termsOfService"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="https://www.example.com/terms-of-service"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="termsOfService"
                          size="large"
                          type="text"
                          value="https://www.example.com/terms-of-service"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="host"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="host"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="host"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="host"
                    >
                      host
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="host"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="api.example.com"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="host"
                          size="large"
                          type="text"
                          value="api.example.com"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="basePath"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="basePath"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="basePath"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="basePath"
                    >
                      basePath
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="basePath"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="/"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="basePath"
                          size="large"
                          type="text"
                          value="/"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="schemes"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="schemes"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="schemes"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="schemes"
                    >
                      schemes
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="schemes"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="https"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="schemes"
                          size="large"
                          type="text"
                          value="https"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="produces"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="produces"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="produces"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="produces"
                    >
                      produces
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="produces"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="application/json"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="produces"
                          size="large"
                          type="text"
                          value="application/json"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
      <Connect(FormItem)
        name="consumes"
      >
        <FormItem
          info={
            Object {
              "basePath": "/",
              "consumes": "application/json",
              "description": "Restful API for Example",
              "host": "api.example.com",
              "produces": "application/json",
              "schemes": "https",
              "termsOfService": "https://www.example.com/terms-of-service",
              "title": "Example API",
              "version": "1.0",
            }
          }
          name="consumes"
          setProp={[Function]}
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="consumes"
            labelCol={
              Object {
                "sm": Object {
                  "span": 6,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
            prefixCls="ant-form"
            wrapperCol={
              Object {
                "sm": Object {
                  "span": 12,
                },
                "xs": Object {
                  "span": 24,
                },
              }
            }
          >
            <Row
              className="ant-form-item"
              gutter={0}
            >
              <div
                className="ant-row ant-form-item"
              >
                <Col
                  className="ant-form-item-label"
                  sm={
                    Object {
                      "span": 6,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-label ant-col-xs-24 ant-col-sm-6"
                  >
                    <label
                      className=""
                      htmlFor={null}
                      title="consumes"
                    >
                      consumes
                    </label>
                  </div>
                </Col>
                <Col
                  className="ant-form-item-control-wrapper"
                  sm={
                    Object {
                      "span": 12,
                    }
                  }
                  xs={
                    Object {
                      "span": 24,
                    }
                  }
                >
                  <div
                    className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-12"
                  >
                    <div
                      className="ant-form-item-control "
                    >
                      <Input
                        disabled={false}
                        onChange={[Function]}
                        placeholder="consumes"
                        prefixCls="ant-input"
                        size="large"
                        type="text"
                        value="application/json"
                      >
                        <input
                          className="ant-input ant-input-lg"
                          disabled={false}
                          onChange={[Function]}
                          onKeyDown={[Function]}
                          placeholder="consumes"
                          size="large"
                          type="text"
                          value="application/json"
                        />
                      </Input>
                    </div>
                  </div>
                </Col>
              </div>
            </Row>
          </FormItem>
        </FormItem>
      </Connect(FormItem)>
    </div>
  </Info>
</Provider>
`;
